vault/seal_autoseal.go

```diff
@@ -430,6 +493,35 @@ func (d *autoSeal) getRecoveryKeyInternal(ctx context.Context) ([]byte, error) {
 	return pt, nil
 }
 
+func (d *autoSeal) upgradeRecoveryKey(ctx context.Context) error {
+	pe, err := d.core.physical.Get(ctx, recoveryKeyPath)
+	if err != nil {
+		return errwrap.Wrapf("failed to fetch recovery key: {{err}}", err)
+	}
+	if pe == nil {
+		return fmt.Errorf("no recovery key found")
```

Comment:
In the other function you return nil here because we may not be initialized. Should these two functions have the same logic? Supposedly these wouldn't currently be called if we are not initialized.

Reply:
Good point, I will fix it.